cloudbuild.yaml

```diff
@@ -74,70 +99,95 @@ steps:
   - PARQUET_PATH=/workspace/e2e-tests/FHIR_SEARCH_HAPI
-  - SINK_PATH=http://sink-server:8080/fhir
+  - SINK_PATH=http://sink-server1:8080/fhir
   - SINK_USERNAME=hapi
   - SINK_PASSWORD=hapi
+  waitFor: ['Build Pipeline Images', 'Upload to HAPI']
 
 - name: '${_REPOSITORY}/e2e-tests:${_TAG}'
   id: 'Run E2E Test for FHIR-search mode with HAPI source'
   env:
   - PARQUET_SUBDIR=FHIR_SEARCH_HAPI
   - DOCKER_NETWORK=--use_docker_network
+  - SINK_SERVER=http://sink-server1:8080
+  waitFor: ['Run Batch Pipeline in FHIR-search mode with HAPI source', 'Build E2E Image']
 
 - name: '${_REPOSITORY}/batch-pipeline:${_TAG}'
   id: 'Run Batch Pipeline for JDBC mode with HAPI source'
   env:
     - JDBC_MODE_ENABLED=true
     - JDBC_MODE_HAPI=true
     - FHIR_SERVER_URL=http://hapi-server:8080/fhir
-    - SINK_PATH=http://sink-server:8080/fhir
+    - SINK_PATH=http://sink-server2:8080/fhir
     - SINK_USERNAME=hapi
     - SINK_PASSWORD=hapi
     - FHIR_DATABASE_CONFIG_PATH=/workspace/utils/hapi-postgres-config.json
     - PARQUET_PATH=/workspace/e2e-tests/JDBC_HAPI
     - JDBC_FETCH_SIZE=1000
+  waitFor: ['Build Pipeline Images', 'Upload to HAPI']
 
 - name: '${_REPOSITORY}/e2e-tests:${_TAG}'
   id: 'Run E2E Test for JDBC mode with HAPI source'
   env:
     - PARQUET_SUBDIR=JDBC_HAPI
     - DOCKER_NETWORK=--use_docker_network
+    - SINK_SERVER=http://sink-server2:8080
+  waitFor: ['Run Batch Pipeline for JDBC mode with HAPI source', 'Build E2E Image']
 
-# The `views` database is used for creating flat views from ViewDefinitions.
+# Resetting FHIR sink server
+- name: 'docker/compose'
+  id: 'Turn down FHIR Sink Server 1'
+  args: [ '-f', './docker/sink-compose.yml', '-p', 'sink-server1', 'down' ,'-v']
+  env:
+    - SINK_SERVER_NAME=sink-server1
+    - SINK_SERVER_PORT=9001
+  waitFor: ['Run E2E Test for FHIR-search mode with HAPI source']
+
+- name: 'docker/compose'
+  id: 'Turn down FHIR Sink Server 2'
+  args: [ '-f', './docker/sink-compose.yml', '-p', 'sink-server2', 'down' ,'-v']
+  env:
+    - SINK_SERVER_NAME=sink-server2
+    - SINK_SERVER_PORT=9002
+  waitFor: ['Run E2E Test for JDBC mode with HAPI source']
+
+  # The `views` database is used for creating flat views from ViewDefinitions.
 - name: 'postgres'
   id: 'Create views database'
   entrypoint: psql
   env:
-  - PGPASSWORD=admin
+    - PGPASSWORD=admin
   args: [ '-U', 'admin', '-d', 'postgres', '-h', 'hapi-fhir-db', '-p', '5432',
           '-c', 'CREATE DATABASE views;']
-
-# Resetting FHIR sink server
-- name: 'docker/compose'
-  id: 'Turn down FHIR Sink Server'
-  args: [ '-f', './docker/sink-compose.yml', 'down' ,'-v']
+  waitFor: ['Turn down FHIR Sink Server 1']
 
 - name: 'docker/compose'
-  id: 'Launch HAPI FHIR Sink Server'
-  args: [ '-f', './docker/sink-compose.yml', 'up','--force-recreate', '-d' ]  
+  id: 'Launch HAPI FHIR Sink Server 1'
+  args: [ '-f', './docker/sink-compose.yml', '-p', 'sink-server1', 'up','--force-recreate', '-d' ]
```

Comment:
Also here, is there any advantage for using the same names? Maybe we can use `sink-server-controller` here? Also is there any significant disadvantage with bringing up all of these sink servers at the beginning (instead of waiting for one to be turned down and then bringing up another one)? IIUC bring up all sink servers at the same time (with different names) should speed up the overall build, no?

Reply:
Changed to have the same `sink-server-controller`. Sometimes, there was resource crunch while bringing multiple servers at the same time. Hence, few services are brought in when they are needed. There should be some benefit when bringing multiple servers at the same time, but we need to optimize on the resources (probably run tests on a higher configuration machine).